tools/haskell-language-server/default.nix

```diff
@@ -1,8 +1,19 @@
-{ fetchFromGitHub,
-  sources }:
+{ fetchFromGitHub
+, sources
+}:
 
-ghcVersion: import sources.nix-hls {
-  inherit ghcVersion;
+{ project
+, # We can't infer this, which is fine as long as we're using a hashed
+  # one.  We're just overriding nix-hls' pin here so haskell.nix looks
+  # it up in its own indexStateHashesPath.
+  index-sha256 ? null
+}:
+let
+  ghcVersion = project.project.pkg-set.options.compiler.nix-name.value;
+  index-state = project.project.index-state;
```

Comment:
We could ask for `earnestProject.project` instead of `project`, and get rid of the `project.project`s here.  We're asking for more than we need, which is bad, but I think this is a more intuitive API for most people.  And this whole thing is just a convenience until it finds its way into haskell.nix's own tooling infrastructure.

Reply:
Yeah, that sounds good!